custom/enikshay/management/commands/drtb_2_to_2_dot_1_migration.py

```diff
@@ -0,0 +1,287 @@
+
+                    if "test" in case_types:
+                        if not occurrence_cases:
+                            occurrence_cases = [case for case in
+                                                accessor.get_reverse_indexed_cases([person_case_id])
+                                                if case.type == CASE_TYPE_OCCURRENCE]
+                        for occurrence_case in occurrence_cases:
+                            test_cases = [case for case in
+                                          accessor.get_reverse_indexed_cases([occurrence_case.get_id])
+                                          if case.type == CASE_TYPE_TEST]
+                            recently_opened_episode_case = None
+                            for test_case in test_cases:
+                                # For tests that are DSTB follow up tests, rft_general = follow_up_dstb
+                                # and result_recorded = yes
+                                recently_opened_episode_case, case_status = self.get_case_status(
+                                    test_case, recently_opened_episode_case=recently_opened_episode_case,
+                                    occurrence_case=occurrence_case)
+                                case_status = self.update_case(test_case.get_id, case_status)
+                                writer.writerow(case_status)
+
+    def update_case(self, case_id, case_status):
+        """
+        :param case_id: id of case to be updated
+        :param case_status: case update status which has updates dict and other information about case update
+        :return: update case status
+        """
+        case_updates = case_status.get('updates')
+        if case_updates:
+            case_updates['datamigration_20_to_21'] = 'yes'
+            if not self.dry_run:
+                update_case(DOMAIN, case_id, case_updates)
+            case_status['updated'] = 'yes'
+        else:
+            case_status['updated'] = 'no'
+        return case_status
+
+    def _get_headers(self, case_types):
+        headers = ['case_type', 'case_id']
+        if 'person' in case_types:
+            headers += (self.person_case_relevant_props + ['owner_id'])
+        if 'episode' in case_types:
+            headers += self.episode_case_relevant_props
+        if 'test' in case_types:
+            headers += self.test_case_relevant_props
+        headers += ['updates', 'updated']
+        return headers
+
+    def get_case_status(self, case, episode_cases=None, recently_opened_episode_case=None, occurrence_case=None):
+        if case.type == "person":
+            return self.get_person_case_status(case)
+        elif case.type == "episode":
+            return self.get_episode_case_status(case, episode_cases)
+        elif case.type == "test":
+            return recently_opened_episode_case, self.get_test_case_status(
+                case, recently_opened_episode_case, occurrence_case)
```

Comment:
You could also just call these functions directly in the `handle_method`, since you always know what type of case you're dealing with.

Reply:
Also, I swear I already reviewed this PR - was there another one recently that shared a lot of code, or did I just not submit the review?

Reply:
The other PR is https://github.com/dimagi/commcare-hq/pull/18027.
This is but one time and the other one would eventually become a reconciliation task that runs more regularly.

Reply:
> You could also just call these functions directly in the handle_method, since you always know what type of case you're dealing with.

Just wanted to extract this logic out so kept it in a method